src/components/PopupConfirmation.js

```diff
@@ -0,0 +1,32 @@
+import Popup from "./Popup.js";
+
+export default class PopupConfirmation extends Popup {
+  constructor(popupSelector, handleConfirm) {
+    super({ popupSelector });
+    this._handleConfirm = handleConfirm;
+    this._confirmButton = this._popupElement.querySelector(
+      ".modal__button-delete"
+    );
+  }
+
+  open(cardId) {
+    this._cardId = cardId; // store cardId when popup opens
+    super.open();
+  }
+
+  setConfirmCallback(callback) {
+    this._handleConfirm = callback;
+  }
+
+  close() {
+    super.close();
+  }
```

Comment:
You shouldn’t create the **same methods** of the parent class if you don’t change them inside and only call them with `super`. The parent methods are available in any inheritor of the class.